pkg/cmd/roachtest/tests/mixed_version_schemachange.go

```diff
@@ -51,7 +51,14 @@ func runSchemaChangeMixedVersions(
 	ctx context.Context, t test.Test, c cluster.Cluster, maxOps int, concurrency int,
 ) {
 	numFeatureRuns := 0
-	mvt := mixedversion.NewTest(ctx, t, t.L(), c, c.All(), mixedversion.NumUpgrades(1))
+	mvt := mixedversion.NewTest(
+		ctx, t, t.L(), c, c.All(),
+		mixedversion.NumUpgrades(1),
+		// Always use latest predecessors, since mixed-version bug fixes only
+		// appear in the latest patch of the predecessor version.
+		// See: https://github.com/cockroachdb/cockroach/issues/121411.
```

Comment:
One example is #122274 - it's fixed on release-23.1 branch but the fix is not present in any 23.1.x releases yet.

Reply:
https://github.com/cockroachdb/cockroach/pull/121237 is a change that would fix the error `could not validate enum value removal`, but is not yet in a 23.2 release. I made this PR pre-emptively so that the test will be less flaky once that PR appears in a 23.2 release.

In general, my team doesn't have the bandwidth to track down too many test flakes. The instant we discover a bug for which we need to backport a fix to an existing release, there's no benefit to testing with `AlwaysUseLatestPredecessors=false`, since it will just result in more chances for noise. (And IMO, it's a fair assumption that there always will be at least a few such bugs in every release.)

Reply:
Thanks for the pointers! Yeah, unfortunately that option won't help for fixes that are in unpublished releases.

> In general, my team doesn't have the bandwidth to track down too many test flakes

Yep, we're all trying to reduce flakes too. It wouldn't be good if teams had (or spent) a lot of time looking at flakes 😄 I was asking mostly because I had trouble figuring out exactly what issue had already been fixed in a latest release. The issue linked in this PR is pretty long, so it's not easy to understand what's going on.

> And IMO, it's a fair assumption that there always will be at least a few such bugs in every release

Agreed. The assumption with that option is that some (hopefully most) of those bugs would be novel because we didn't use to have good upgrade testing coverage. That said, I think this option has caused a lot more trouble than actually helped find bugs. I'll consider removing it altogether or at least changing the default.

Maybe providing an option to use a release branch directly (instead of latest published release) could be useful too.